sdk/core/core/inc/_az_cfg.h

```diff
@@ -91,4 +91,8 @@
 // Get the number of elements in an array
 #define _az_COUNTOF(array) (sizeof(array) / sizeof(array[0]))
 
+#ifndef UNUSED
+#define UNUSED(x) (void)(x)
```

Comment:
Just as a note: this doesn't work in GCC.
[Link to GCC discussion](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425).

One way we discussed doing this is using `assert()`

Reply:
It worked with GCC since that's what's used in the build system and on my box. Should work with clang too since this is what we're using in C-SDK in a lot of places.

Can you give a short explanation/TLDR on what exactly won't work? (I'm, thinking increasing the warning level - didn't try that.)

Reply:
Shoot sorry I take that back. It works for input parameters but it does NOT work for returned values of functions. So you can't void cast a returned value from one of our functions and silence that warning.